src/router/index.js

```diff
@@ -6,6 +6,19 @@ const { routeConfigInjector } = require('@middlewares/routeConfigInjector')
 const { rateLimiter } = require('@middlewares/rateLimiter')
 const bodyParser = require('body-parser')
 const { httpMethods } = require('@constants/httpMethods')
+// Function to handle JSON parsing errors
+const handleJsonParsingError = (req, res, next) => {
+    bodyParser.json({ limit: '50MB' })(req, res, err => {
+        if (err instanceof SyntaxError) {
+            res.status(400).json({
+                responseCode: 'INVALID_JSON', // response code and message can be customized here
+                message: 'The JSON provided in the request is invalid or malformed.',
+            });
+        } else {
+            next(err);
+        }
+    });
+};
```

Comment:
rename this as jsonBodyParserWithError and move it to middleware folder and import it into this file like other middlewares.